src/inmanta/execute/scheduler.py

```diff
@@ -192,17 +208,20 @@ def define_types(self, compiler, statements, blocks):
 
         self.types = {k: v for k, v in types_and_impl.items() if isinstance(v, Type)}
 
-    def anchormap(self, compiler, statements, blocks) -> List[Tuple[Location, Location]]:
+    def anchormap(
+        self, compiler: "Compiler", statements: Sequence["Statement"], blocks: Sequence["BasicBlock"]
+    ) -> Sequence[Tuple[Location, Location]]:
         prev = time.time()
 
         # first evaluate all definitions, this should be done in one iteration
         self.define_types(compiler, statements, blocks)
 
         # relations are also in blocks
-        statements = (s for s in statements if not isinstance(s, DefineRelation))
-        anchors = (
+        not_relation_statements: Generator[Statement, Any, Any] = (s for s in statements if not isinstance(s, DefineRelation))
+
+        anchors: Generator[Anchor, Any, Any] = (
```

Comment:
```suggestion
        not_relation_statements: Generator[Statement, None, None] = (s for s in statements if not isinstance(s, DefineRelation))

        anchors: Generator[Anchor, None, None] = (
```
From [the docs](https://docs.python.org/3/library/typing.html#typing.Generator): "If your generator will only yield values, set the `SendType` and `ReturnType` to `None` ... Alternatively, annotate your generator as having a return type of ... `Iterator[YieldType]`"

Reply:
thx, updated.